homeassistant/components/lg_netcast/media_player.py

```diff
@@ -54,6 +55,7 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_ON_ACTION): cv.SCRIPT_SCHEMA,
+        vol.Optional(CONF_OFF_ACTION): cv.SCRIPT_SCHEMA,
```

Comment:
We should add an option for this since there's already a native turn off function for this tv. A custom script can handle other cases.

Reply:
Sry, i did not get your point. There is a purpose to override the native `turn off` function, but you do not have to use this if you are satisfied with the native standby option. But if one likes to handle the power situation for the tv without the native standy options, here the way.
I may rename the option to `custom_turn_off_action` but I'm not sure how this might be confusing in relation to the defineable `turn_on_action`.

Reply:
It's already possible to make an automation or script that is triggered when the TV is turned off. We don't need an option for that here.

Reply:
Okay. But the purpose here is, to trigger an action _instead_ of the native `turn_off`, not in addition

Reply:
Exactly like the `turn_on` (but there was no native `turn_on`). 
With this it's possible to handle the full power with a power plug/socket (which totally fits my need, because i dont have to turn on the power plug and then turn on the tv via remote):
- Runnning TV -> Switch off power plug -> Switch on power plug -> TV autmatically starting.
without
- Running TV -> Switch off TV -> Switch off power plug (automation) -> Switch on power plug -> (TV is off) -> Turn on TV in an extra step (ex. via remote)

Reply:
As long as there is no way to start my LG TV automatically when it was natively turned off and the power plug is turned on, I did not see a way to handle this with the given tools via script or automation. 

(I could implement an IR sender which sends an IR code to turn on the TV if the power plug is switched on, but that looks like a bit overkill hw-setup to me in relation to this 13 additional lines)

Reply:
Our guideline is to only model what each device exposes. The existing turn on script is already a step away from that direction, made as an exception. It's not a precedent to extend with more features that the device doesn't expose natively.

Different devices have different features and behave differently. We can't and shouldn't try to add options and mock features to cater for every possible need. We model the device as we are able from the architecture design we have set.

You're welcome to open an issue in our architecture repo and describe what, why and how we should change our design regarding this.

https://github.com/home-assistant/architecture/

Reply:
Hi Martin, thanks for the clearification, I did not know that. 
But I'm actually going to file an issue in the architecture repo, because I do believe there is a certain value to give this integration (or maybe more abstract all `media_player` devices) the possiblity to behave with more flexibility in context of power switching.